robot/impl/resource_manager.go

```diff
@@ -335,9 +336,16 @@ func (manager *resourceManager) Close(ctx context.Context) error {
 		if !ok {
 			continue
 		}
+
 		if err := utils.TryClose(ctx, iface); err != nil {
 			allErrs = multierr.Combine(allErrs, errors.Wrap(err, "error closing resource"))
 		}
+
+		if r.modules != nil && r.modules.IsModularResource(x) {
```

Comment:
for these functions updates here, I'm wondering if there's any more succinct way to have this be built in to our "type system" such that you reconfigure a resource just the same as anything else. This may be pending the @cheukt @npmenard reconfig discussion.